taichi/backends/vulkan/runtime.cpp

```diff
@@ -427,9 +441,24 @@ void VkRuntime::launch_kernel(KernelHandle handle, RuntimeContext *host_ctx) {
       &ti_kernel->ti_kernel_attribs().ctx_attribs, host_ctx, device_,
       host_result_buffer_, ctx_buffer.get(), ctx_buffer_host.get());
 
+  std::unordered_map<int, DeviceAllocation> ext_arrays;
```

Comment:
FYI in opengl we don't redo device allocation if the size doesn't change. maybe it helps with perf a little bit? https://github.com/taichi-dev/taichi/blob/master/taichi/backends/opengl/opengl_api.cpp#L400 (can be a followup ofc)

Reply:
Let's leave it for later, the balance between memory usage and allocation performance is hard